app/models/concerns/digital_object_concerns/publish_behavior.rb

```diff
@@ -146,22 +148,30 @@ def unpublish_from_all
 
     private
 
+      def publish_promise(&block)
+        Concurrent::Promise.execute(&block)
+      end
+
+      def unpublish_promise(&block)
+        Concurrent::Promise.execute(&block)
+      end
```

Comment:
@barmintor I know this helps with testing, but I think it's a little confusing to have two similarly named methods that do exactly the same thing.  Maybe worth consolidating them into a single method called `execute_promise`?

Reply:
I would have been less likely to make the above suggestion if `unpublish_promise` were used in a test.

Reply:
I would have been less likely to make the above suggestion if `unpublish_promise` was being mocked in a test.

Reply:
It definitely _will_ be used in a test - my recommendation is that we wait to consolidate.

Reply:
Ah, makes sense.  Just had another thought too, and this would be an easy change:

How about we call these `async_publish` and `async_unpublish` and then it would be natural to assume that they return promises, but then we would be naming the methods after what they do and not after their return types?